### PR TITLE
Remove radio free range

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Fixes a typo in Antimov.
2) Ports Bay's radio range overhaul: https://github.com/Baystation12/Baystation12/pull/10934.

From Bay's PR:
*Manual radio frequency changes can no longer go outside the standard frequency span.
*Users with sufficient access can select pre-defined channels outside this span, such as department channels.

### CODE REVIEW

**Comment by Regen1** [comment]
What will this do?
Does this mean I have to change to different departments by clicking my headset in-hand and opening the interface to change the frequency?

**Comment by Markolie** [comment]
No, it means that you can no longer *manually* set the frequency of a radio to whatever you want: it's now limited to a set amount of frequencies. And rather than having to remember the frequency of engineering, command, service, etc. you can now set a radio to that frequency (i.e. station bounced radio) if you simply have the required access.

**Comment by KasparoVy** [comment]
"prevents every random greytider from setting an intercom frequency to AI Private"
Lmfao, that's always fun.
A welcome fix

**Comment by Alex6511** [comment]
Awwww, now I can't secretly communicate with the AI as the clown and tell it to let me out of the brig when I get arrested for slipping the captain.

**Comment by tigercat2000** [comment]
Uh, what? AI private is within the normal public frequency range, how does this touch that?

**Comment by Markolie** [comment]
It's moved to 1343.

**Comment by tigercat2000** [comment]
Oh, alright. I am kindof against moving it there, since it has always been a semi-public frequency, but whatevs I guess.

File: code/game/jobs/access.dm [hunk 1]
```diff
@@ -186,6 +186,17 @@
...
+/proc/has_access(var/list/req_access, var/list/req_one_access, var/list/accesses)
+	for(var/req in req_access)
+		if(!(req in accesses)) //doesn't have this access
+			return 0
+	if(req_one_access.len)
+		for(var/req in req_one_access)
+			if(req in accesses) //has an access from the single access list
+				return 1
+		return 0
+	return 1
```
Comment (PsiOmegaDelta) [inline code/game/jobs/access.dm]: Just a minor thing, but line 180-188 above is basically replaced with a call to this proc.

Comment (Markolie) [inline code/game/jobs/access.dm]: Ah, thank you! Will fix.

**Comment by Earthdivine** [comment]
Alright since I'm a goof and can't understand this correctly, this will NOT remove peoples abilities to set up their own channels, and just redoes some of them so they cant go to certain ranges, like the AI Private?

**Comment by tigercat2000** [comment]
@Earthdivine Basically, it removes the intercomms like the captain's and AI's that can go to any frequency, replacing it with buttons for each freq that the person interacting with it has the 'access level' to, and cleans up some of the code. It also moves a couple channels to different frequencies outside of the public range

**Comment by M3hillus** [comment]
@Markolie The AI should be able to set wallcomms to its private channel still, correct?

**Comment by Spacemanspark** [comment]
What about the pirate DJ station?
Will we still be able to listen to all channels with that?

**Comment by Markolie** [comment]
Yes, the AI and the Captain can set channels to AI Private.

Pirate DJ station will no longer be able to switch to all channels, but I can look into making that possible.

**Comment by Spacemanspark** [comment]
Please do.
Other than that, :+1:.

**Comment by Alex6511** [comment]
Don't forget the syndicate dj station either, its in the lower left of the station.

**Comment by PsiOmegaDelta** [comment]
Simply add to or fully replace the internal_channels list in /New() for a new radio subtype, similar to how /obj/item/device/radio/intercom/syndicate does it, and you should be good.

**Comment by Markolie** [comment]
That works, thank you!

**Comment by Markolie** [comment]
Map editing and testing complete. can be merged.

Would be nice if this could be merged quickly due to merge conflicts (it edits every map file).

**Comment by tigercat2000** [comment]
Um, you did make sure metastation compiles with it, right?

**Comment by Markolie** [comment]
Don't think Metastation compiles at the moment anyway.

**Comment by tigercat2000** [comment]
Probably doesn't. :disappointed:

**Comment by Markolie** [comment]
We should really switch to MetaStation for some variety some day soon (and when that happens it shouldn't take too much effort to make it compile).

**Comment by tigercat2000** [comment]
Knew I should have mentioned that this PR will break metastation's compile beyond my ability to fix, because I don't know everything that needs to be changed, *before* I left for an hour

**Pull request merged**